src/test/java/io/appium/java_client/android/LogEventTest.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.java_client.android;
+
+import io.appium.java_client.driverscripts.ScriptOptions;
+import io.appium.java_client.driverscripts.ScriptType;
+import io.appium.java_client.driverscripts.ScriptValue;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+
+public class LogEventTest extends BaseAndroidTest {
+
+    @Test
+    public void verifyLoggingCustomEvents() {
+        driver.logEvent("appium", "funEvent");
+        Object value = driver.getSessionDetail("events");
```

Comment:
session details is not a part of W3C spec. Should we have events under "mobile:" endpoint?

Reply:
hmm. the problem with the mobile endpoints is that they are supported in individual drivers, not basedriver, i think. i'm not sure right now of a clean way to cascade mobile calls from basedriver to subdrivers. we'd have to pull a lot of logic in from other drivers and clean it up somehow.

Reply:
perhaps, similarly to how we use executeDriverScript

Reply:
@mykola-mokhnach to make this functionality available in w3c, should we create a `GET /session/:sessionid/appium/log_events` method?

Reply:
yes, we would need something like that. Also, the method might require additional filtering parameters, so I don't know if it makes sense to change the method to POST

Reply:
https://github.com/appium/appium-base-driver/pull/365

Reply:
ok, i created the various java objects needed for first class support of that route.